sgkit_vcf/vcf_reader.py

```diff
@@ -0,0 +1,286 @@
+            ds["variant_id_mask"] = (
+                [DIM_VARIANT],
+                variant_id_mask,
+            )
+            ds.attrs["max_variant_id_length"] = max_variant_id_length
+            ds.attrs["max_variant_allele_length"] = max_variant_allele_length
+
+            if first_variants_chunk:
+                # Enforce uniform chunks in the variants dimension
+                # Also chunk in the samples direction
+                encoding = dict(
+                    call_genotype=dict(chunks=(chunk_length, chunk_width, n_ploidy)),
+                    call_genotype_mask=dict(
+                        chunks=(chunk_length, chunk_width, n_ploidy)
+                    ),
+                    call_genotype_phased=dict(chunks=(chunk_length, chunk_width)),
+                    variant_allele=dict(chunks=(chunk_length, n_allele)),
+                    variant_contig=dict(chunks=(chunk_length,)),
+                    variant_id=dict(chunks=(chunk_length,)),
+                    variant_id_mask=dict(chunks=(chunk_length,)),
+                    variant_position=dict(chunks=(chunk_length,)),
+                    sample_id=dict(chunks=(chunk_width,)),
+                )
+
+                ds.to_zarr(output, mode="w", encoding=encoding)
+                first_variants_chunk = False
+            else:
+                # Append along the variants dimension
+                ds.to_zarr(output, append_dim=DIM_VARIANT)
+
+
+def vcf_to_dataset(
+    input: PathType,
+    output: PathType,
+    region: Optional[str] = None,
+    chunk_length: int = 10_000,
+    chunk_width: int = 1_000,
+) -> xr.Dataset:
+    vcf_to_zarr_sequential(input, output, region, chunk_length, chunk_width)
+    ds: xr.Dataset = xr.open_zarr(str(output))  # type: ignore[no-untyped-call]
+    return ds
+
+
+def vcf_to_zarr_parallel(
+    input: Union[PathType, Sequence[PathType]],
+    output: PathType,
+    regions: Union[None, Sequence[str], Sequence[Optional[Sequence[str]]]],
+    chunk_length: int = 10_000,
+    chunk_width: int = 1_000,
+) -> None:
+    """Convert specified regions of one or more VCF files to zarr files, then concat, rechunk, write to zarr"""
+
+    if isinstance(input, str) or isinstance(input, Path):
+        # Single input
+        inputs: Sequence[PathType] = [input]
+        assert regions is not None  # this would just be sequential case
+        input_regions: Sequence[Optional[Sequence[str]]] = [regions]  # type: ignore
+    else:
+        # Multiple inputs
+        inputs = input
+        if regions is None:
+            input_regions = [None] * len(inputs)
+        else:
+            if len(regions) == 0 or isinstance(regions[0], str):
+                raise ValueError(
+                    f"Multiple input regions must be a sequence of sequence of strings: {regions}"
+                )
+            input_regions = regions
+
+    assert len(inputs) == len(input_regions)
+
+    tempdir = Path(tempfile.mkdtemp(prefix="vcf_to_zarr_"))
```

Comment:
This seems worth parameterizing in case a user wants to target a specific volume with greater capacity.